Device can't be created using sycl::*_selector_v

According to the specification a device can be initialized using the one of `sycl::*_selector_v` constructs, this is currently not possible. According to the [specification](https://www.khronos.org/registry/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf) page 56 this should work.

code:
```
#include <sycl/sycl.hpp>

int main() {
  sycl::device gpu{sycl::gpu_selector_v};
}
```

error:
```
/build/nbody_hipsim/src/test.cpp:4:16: error: no matching constructor for initialization of 'sycl::device'
  sycl::device gpu{sycl::gpu_selector_v};
               ^  ~~~~~~~~~~~~~~~~~~~~~~
/nix/store/f1jrjf6b8813dbjal6kj8qkphgn22596-hipsycl/bin/../include/sycl/../hipSYCL/sycl/device.hpp:69:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const hipsycl::sycl::gpu_selector' to 'const hipsycl::sycl::device' for 1st argument
class device {
      ^
/nix/store/f1jrjf6b8813dbjal6kj8qkphgn22596-hipsycl/bin/../include/sycl/../hipSYCL/sycl/device.hpp:69:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const hipsycl::sycl::gpu_selector' to 'hipsycl::sycl::device' for 1st argument
/nix/store/f1jrjf6b8813dbjal6kj8qkphgn22596-hipsycl/bin/../include/sycl/../hipSYCL/sycl/device.hpp:75:3: note: candidate constructor not viable: no known conversion from 'const hipsycl::sycl::gpu_selector' to 'rt::device_id' for 1st argument
  device(rt::device_id id)
  ^
/nix/store/f1jrjf6b8813dbjal6kj8qkphgn22596-hipsycl/bin/../include/sycl/../hipSYCL/sycl/device_selector.hpp:231:16: note: candidate constructor not viable: no known conversion from 'const hipsycl::sycl::gpu_selector' to 'const hipsycl::sycl::device_selector' for 1st argument
inline device::device(const device_selector &deviceSelector) {
               ^
/nix/store/f1jrjf6b8813dbjal6kj8qkphgn22596-hipsycl/bin/../include/sycl/../hipSYCL/sycl/device.hpp:78:3: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
  device()
  ^
1 error generated when compiling for gfx906.
```

The specification is in my opinion very loosely formulated, so I have no idea what should be changed.

--- a/include/hipSYCL/sycl/device.hpp
+++ b/include/hipSYCL/sycl/device.hpp
@@ -79,7 +79,8 @@ public:
       : _device_id(detail::get_host_device()) {}
 
   // Implemented in device_selector.hpp
-  explicit device(const device_selector &deviceSelector);
+  template <class DeviceSelector>
+  explicit device(const DeviceSelector &deviceSelector);
 
   bool is_host() const 
   {

--- a/include/hipSYCL/sycl/device_selector.hpp
+++ b/include/hipSYCL/sycl/device_selector.hpp
@@ -228,8 +228,9 @@ auto aspect_selector() {
   return aspect_selector(aspectList...);
 }
 
-inline device::device(const device_selector &deviceSelector) {
-  this->_device_id = deviceSelector.select_device()._device_id;
+template <class DeviceSelector>
+inline device::device(const DeviceSelector &deviceSelector) {
+  this->_device_id = detail::select_device(deviceSelector)._device_id;
 }
 
 

